tools/toolchains/__init__.py

```diff
@@ -188,6 +188,23 @@ def __str__(self):
 }
 
 
+def check_toolchain_path(function):
+    """Check if the path to toolchain is valid. Exit if not.
+    Use this function as a decorator.  Causes a system exit if the path does
+    not exist. Execute the function as normal if the path does exist.
+
+    Positional arguments:
+    function -- the function to decorate
+    """
+    def perform_check(self, *args, **kwargs):
+        if not exists(self.toolchain_path):
+            error_string = 'Could not find executable for %s.\n Currently ' \
```

Comment:
If I'm reading this right, the error string is a bit misleading. You're not checking for an executable, but rather for the top directory of the whole toolchain, is that right?